### PR TITLE
Replace `expectedAlertNondeterministic` with simpler check function

### LINKED ISSUE #84807: Replace nondeterministic alert test decorator `expectedAlertNondeterministic` with simpler function
It has been bugging me for a while that the `expectedAlertNondeterministic` decorator used for nondeterministic alert tests in `test/test_torch.py` is more complicated and confusing than it needs to be. If I remember correctly, it used to really need to be a decorator, but we don't need it to be a decorator any longer, so I would like to replace it with a simpler nondeterministic alert testing function that just runs a given lambda or function and checks that it produces the alert in the expected cases. This should make writing and reading the code for these tests easier.

cc @mruberry

### PR DESCRIPTION
**Pull request by kurtamohler**
Fixes #84807

### CODE REVIEW

File: torch/testing/_internal/common_utils.py [hunk 1]
```diff
@@ -2673,6 +2673,55 @@ def assertAtenOp(self, onnx_model, operator, overload_name=""):
...
+        # Check that errors are thrown correctly
+        with DeterministicGuard(True):
+            if should_alert:
+                with self.assertRaisesRegex(
+                        RuntimeError,
+                        error_message,
+                        msg='expected a non-deterministic error, but it was not raised'):
+                    fn()
+
+            else:
```
Comment (mruberry) [inline torch/testing/_internal/common_utils.py:2704]: Interesting lack of parallelism between this if/else block and the following for warnings. Why handle the error not thrown case here but not the warning not thrown case?

**Review by mruberry — APPROVED** [review]
If you prefer this let's go for it. One inline note for your consideration, @kurtamohler. It's not a regression and doesn't need to be addressed, but interesting to consider while we're here.

File: torch/testing/_internal/common_utils.py (hunk 1 above)
Comment (kurtamohler) [reply torch/testing/_internal/common_utils.py:2704]: Good point, I'll add the not thrown case

Comment (kurtamohler) [reply torch/testing/_internal/common_utils.py:2704]: I've added it

**Pull request closed without merging**